msteams-platform/m365-apps/specify-runtime-requirements.md

```diff
@@ -0,0 +1,142 @@
+
+### One-way dependencies
+
+Use the `oneWayDependencies` array to describe cases where one component of your app depends upon another component. For each object in the array, specify the dependent component (`element`) and the component it depends on (`dependsOn`). For example:
+
+```json
+    "elementRelationshipSet": {
+      "oneWayDependencies" : [
+        {
+          "element" : {
+            "name" : "composeExtensions",
+            "id" : "composeExtension-id",
+            "commandIds": ["command-1-id", "command-2-id"]  // Developers can add more commands.
+          },
+          "dependsOn" : [
+              {"name" : "bots", "id" : "bot-id"}
+            ]
+        }
+      ]
+    }
+```
+
+For message extensions, you can optionally specify individual commands that require support for specific app components. If those components aren't supported in the runtime host, they will not be made available to the end-user (though all other commands will run).
+
+### Mutual dependencies
+
+Use the `mutualDependencies` array to group app components that must load together in order to support their intended function. Each object in the array represents a mutually dependent app component. For example:
+
+```json
+    "elementRelationshipSet": {
+      "mutualDependencies" : [
+                {"name" : "bots", "id" : "bot-id"}, 
+                {"name" : "staticTabs", "id" : "static-Tab-id"},
+                {"name" : "composeExtensions", "id" : "composeExtension-id"},
+      ]
+    },
+```
+
+## Specify runtime capability requirements for specific app components (`requirementSet`)
+
+When you're defining an individual app component, you can specify it's specific TeamsJS runtime requirements using a `requirementSet`. This will ensure the component only loads in Microsoft 365 hosts with support for the critical TeamsJS capabilities. For example:
+
+```json
+    "staticTabs": [
+        {
+            "entityId": "idForPage",
+            "name": "Display name of tab",
+            "contentUrl": "https://contoso.com/content?host=msteams",
+            "contentBotId": "Specifies to the app that tab is an Adaptive Card Tab. You can either provide the contentBotId or contentUrl.",
+            "websiteUrl": "https://contoso.com/content",
+            "scopes": [
+                "personal"
+            ],
+            "requirementSet": {
+                "hostMustSupportFunctionalities": [
+                  {"name": "dialogUrl"},
+                  {"name": "dialogUrlBot"}
+                ]
+            }
+        }
+    ],
+```
+
+Currently, a subset of TeamsJS capabilities can be specified as runtime requirements for individual components of an app. This support will expand over time. The following TeamsJS capabilities can be specified as runtime requirements for `staticTabs`, `composeExtensions`, and `bots`:
+
+- HTML-based dialogs (`dialog.url`)
```

Comment:
Just gentle reminder, but I suppose it'll be part of your TODO Erika to add cross-referencing to these capabilities. :)

Reply:
The new sections in manifest reference already point to RT article, and I've just pushed an update w/links from RT article to manifest reference.